src/net_processing.cpp

```diff
@@ -2465,6 +2466,7 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
 
         // Feeler connections exist only to verify if address is online.
         if (pfrom.IsFeelerConn()) {
+            LogPrint(BCLog::NET, "feeler connection completed peer=%d; disconnecting\n", pfrom.GetId());
```

Comment:
in commit 9c247a74f0e6b448f9c879c41956e318c753783a:

Could use the helper to avoid having to change the string here if feeler connections get renamed in any way?

```suggestion
            LogPrint(BCLog::NET, "%s connection completed peer=%d; disconnecting\n", pfrom.ConnectionTypeAsString(), pfrom.GetId());
```

Reply:
It's being disconnected because it matches `IsFeelerConn`, which is "hardcoded" above -- matching that seems right to me